File: Program/LawCreator/test/test/lawcreator/xml/nodes/NodeAttributeTest.java
package test.lawcreator.xml.nodes;

import static org.junit.Assert.*;

import org.junit.Before;
import org.junit.Test;

import pl.wroc.pwr.student.lawcreator.xml.nodes.Attribute;

public class NodeAttributeTest {
	Attribute attribute;

	@Before
	public void setUp() throws Exception {
		attribute = new Attribute();
	}

	@Test
	public void gettingAndSettingFieldsTest() {
		String name = "name";
		String value = "value";
		
		attribute.setName(name);
		attribute.setValue(value);
		
		assertEquals(name, attribute.getName());
		assertEquals(value, attribute.getValue());
	}
	
	@Test
	public void gettingXMLTest() {
		String name = "name";
		String value = "value";
		
		attribute.setName(name);
		attribute.setValue(value);
		
		assertEquals(name + "=\"" + value + "\"", attribute.toXML());
	}

}

File: Program/LawCreator/bin/test/lawcreator/xml/nodes/NodeAttributeTest.js
Clazz.declarePackage ("test.lawcreator.xml.nodes");
Clazz.load (null, "test.lawcreator.xml.nodes.NodeAttributeTest", ["pl.wroc.pwr.student.lawcreator.xml.nodes.Attribute"], function () {
c$ = Clazz.decorateAsClass (function () {
this.attribute = null;
Clazz.instantialize (this, arguments);
}, test.lawcreator.xml.nodes, "NodeAttributeTest");
Clazz.defineMethod (c$, "setUp", 
function () {
this.attribute =  new pl.wroc.pwr.student.lawcreator.xml.nodes.Attribute ();
});
Clazz.defineMethod (c$, "gettingAndSettingFieldsTest", 
function () {
var name = "name";
var value = "value";
this.attribute.setName (name);
this.attribute.setValue (value);
org.junit.Assert.assertEquals (name, this.attribute.getName ());
org.junit.Assert.assertEquals (value, this.attribute.getValue ());
});
Clazz.defineMethod (c$, "gettingXMLTest", 
function () {
var name = "name";
var value = "value";
this.attribute.setName (name);
this.attribute.setValue (value);
org.junit.Assert.assertEquals (name + "=\"" + value + "\"", this.attribute.toXML ());
});
});
